### PR TITLE
[AMD][CI] Fix entrypoints pooling cross-encoder vision test group on gfx942

### PR DESCRIPTION
**Pull request by aarushjain29**
## Purpose

The `mi300_1-entrypoints-integration-pooling` is failing on mi300. Five tests in `entrypoints/pooling/scoring/test_cross_encoder_online_vision.py` fail, all on `TRITON_ATTN`, all the same `TEXT_VS_TEXT` assertion for `Qwen/Qwen3-VL-Reranker-2B`:

[TRITON_ATTN] text_vs_text: actual=0.108373 expected=0.100404 rel_diff=0.0794 tol=0.045 abs_tol=0.0

## Root cause

Numerical drift on a near-zero softmax probability, **not** a functional/kernel bug:

- The score is `sigmoid(logit_yes − logit_no)`. The drift is ~0.085 nats (~3.9% in logit space); the sigmoid slope at `p≈0.1` amplifies an ordinary ~0.008 bf16 absolute difference into ~7.9% **relative**.
- An independent backend reproduces the value: `ROCM_ATTN` math attention yields `0.108287` vs Triton's `0.108373` 
A Triton kernel bug could not be reproduced by a different attention path. **`ROCM_ATTN` passes only because its relative tol is already `0.09`.**
- Deterministic across runs/endpoints; ranking is preserved (irrelevant pair ~0.108 ≪ relevant 0.54 / 0.74).
- The stale `gfx942:~2.20%` comment was measured on a larger score, so the bound was set too tight for this lowest-probability pair.

Per-backend `text_vs_text` on the same gfx942 GPU:

| backend | actual | abs diff | tol / abs_tol | result (before) |
|---|---|---|---|---|
| ROCM_ATTN | 0.108287 | 0.00788 | 0.09 / 0.0 | pass |
| TRITON_ATTN | 0.108373 | 0.00797 | 0.045 / 0.0 | **FAIL** |
| ROCM_AITER_FA | 0.103778 | 0.00338 | 0.045 / 0.005 | pass |
| FLEX_ATTENTION | 0.102054 | 0.00165 | 0.045 / 0.006 | pass |

## Solution

Add a small absolute-tolerance floor `"TRITON_ATTN": 0.009` to `BACKEND_ABS_TOL`, mirroring the existing floors for `ROCM_AITER_FA` (0.005) and `FLEX_ATTENTION` (0.006). 

The value equals the absolute drift the suite already permits for `ROCM_ATTN` on this pair (`0.09 × 0.1004 ≈ 0.009`) and covers the observed `0.00797` with ~13% margin. The 4.5% relative tolerance is unchanged, so the larger, ranking-relevant scores stay strictly checked. 

 gfx942 TRITON/ROCM_ATTN drift roughly doubled under ROCm 7.2 and TRITON converged onto the ROCM_ATTN value (likely a kernel fallback), tracked in https://github.com/vllm-project/vllm/issues/35569; tolerance relaxed temporarily pending the kernel fix.

## Test Plan

```bash
cd /vllm-workspace/tests
pytest -v -s entrypoints/pooling/scoring/test_cross_encoder_online_vision.py


Test Result
Before: 5 failed, 6 passed (TRITON_ATTN) — actual=0.108373, rel_diff=7.94% > tol=4.5%, abs_tol=0.0.

====== 5 failed, 6 passed, 33 deselected, 16 warnings in 80.01s (0:01:20) ======
After: 11 passed (TRITON_ATTN) and 44 passed (all four backends). Same model output 0.108373 now within abs_tol=0.009; all other comparisons unchanged.


========== 11 passed, 33 deselected, 16 warnings in 77.71s (0:01:17) ===========
================= 44 passed, 16 warnings in 303.86s (0:05:03) ==================

### CODE REVIEW

**Comment by aarushjain29** [comment]
Related issue - https://github.com/vllm-project/vllm/issues/35569

**Comment by AndreasKaratzas** [comment]
Let's close this as it was fixed elsewhere.
